tests/components/http/test_request_context.py

```diff
@@ -0,0 +1,33 @@
+"""Test request context middleware."""
+from contextvars import ContextVar
+
+from aiohttp import web
+
+from homeassistant.components.http.request_context import setup_request_context
+
+
+async def test_request_context_middleware(aiohttp_client):
+    """Test that request context is set from middleware."""
+    context = ContextVar("request", default=None)
+    app = web.Application()
+
+    async def mock_handler(request):
+        """Return the real IP as text."""
+        request_context = context.get()
+        assert request_context
+        assert request_context == request
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)